\documentclass[UKenglish,11pt,a4paper]{article}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc, url}
\usepackage{graphicx,babel,csquotes}
\usepackage[backend=biber,style=numeric-comp,sorting=nyt]{biblatex}
\addbibresource{macros.bib}
\addbibresource{references.bib}
\title{Ranklust - A bioinformatics solution to identify network biomarkers in cancer}
\author{Henning Lund-Hanssen}
\begin{document}
\maketitle
\tableofcontents{}
\part{Introduction}
To this day we still struggle with cancer. Even with all our modern equipment and knowledge we have still not been able
to tame this horrible disease. My thesis is about making a tool which gives cancer researchers an easier way of
identifying network biomarkers in cancer. 
\section{State of today}
\subsection{Biomarkers}
Biomarkers are at the centre of this thesis. They are what Ranklust should be able to detect and rank in order to
identify network biomarkers, and not just single molecules of them. 
\subsection{Prostate specific antigen}
An example of a single molecule biomarker is the prostate specific antigen. This is 


Today it is used for both identifying and
evaluating the current stage of prostate cancer. This biomarker can be found by analyzing blood examples from patients,
thus fulfilling some of the demands for a good biomarker, %citation needed!!
but not all of them. It is easy to measure and easy to acquire, but not reliable enough to be used as the only tool 
% find a better word, "tool" sucks!
to identify and determine the stage or remission of prostate cancer.
\subsection{Next-generation sequencing}
\section{Single gene prioritization}
\section{Network generation}
\section{Clustering}
\section{Cytoscape}
% Rename the Cytoscape section to Software/Architecture/Development
% General description of development
Cytoscape is the open-source software platform the Ranklust app will be developed on. Its main purpose is to visualize
molecular interaction networks and biological pathways. It has an easy way to integrate \textbf{\textit{Apps}}, which may
be combined by other apps again to build big and complex applications which may solve problems in a bigger picture. The
goal of Ranklust is to cluster the networks we get from single gene prioritization and rank them in order to identify
network biomarkers. Apps taking care of making the networks already exists, but there still has to be made a decision
about whether or not they could be modified in order to better support the clustering.

% Databases
Which databases to use has to be considered. The reason to use databases is because they have information on how protein 
and genes form a network based on how they interact with eachother. The initial database candidates in Ranklust are
iRefIndex \cite{iri}, GeneMania \cite{gm} and STRING \cite{str}. These databases all have in common that it exists
Cytoscape apps made to use these databases. STRING however, does not have any repository available through the Cytoscape
app store, so interacting with the database through a new app in Cytoscape without making new plugins may be difficult.
On the other hand, both iRefIndex and GeneMania have their repositories easily available to the public together with
decent documentation. However, the difference between them is what they contain information about. iRefIndex contains
data about protein-protein interaction (PPI), while GeneMania contains data about genes.
% Check this next one out! Link from source!
Since proteins come from genes, GeneMania can also give us some information about proteins. Differences between the
two databases will be discussed in greater depth at a later stage.

The open-source plugins in Cytoscape to communicate with the databases are iRefScape \cite{iridb} for iRefIndex and 
GeneMANIA \cite{gmdb} for GeneMania.

% Cytoscape
Cytoscape is based on the Java programming language, which is a little bit untraditional for a software platform used to
develop bioinformatics tools. % cite needed!!!
The reason to choose Java above Python, Perl or other popular programming languages is simply because I am more
versed in Java programming than any other language. The Cytoscape community also promotes the idea of developing apps
that follow the \emph{OSGi} standard \cite{cytoscape-osgi}.

% OSGi and design patterns
Developing OSGi software should promote modularization % cite needed!!!
of the code and increase the probability of the app being launched as an official Cytoscape app, in addition to provide
other developers with the possibility of reusing my modules in their own apps. There exists several design patterns that
could prove to be useful in the development of Ranklust. Another strategy to follow may not be a direct design patter,
but more of a collection of them, is the clean code principles. More thorough examination of these strategies will
follow. % Cite needed for design pattern definition and clean code principles

% Prototyp section
Prototyping of different parts of the app will be done in Python and the Galaxy environment \cite{galaxy}. Python is an 
easy language to prototype with, and Galaxy is an easy environment to test small scripts with. Galaxy also has great 
logging of previous experiments combined with settings, so recreating simulations and comparing results is easy to do and
reliable. However, in my experience, Java comes up to par with Python in development speed once all of the boilerplate
code is written and a good deployment tool is used. Therefore the Cytoscape platform is used for the final deployment of
Ranklust.

% Maven section

\part{Background}
\section*{Introduction}
We have our body, and inside our body we have our organs. These organs are made up of tissue, and tissue is made up of
cells. Our cells performs two type of functions, to execute chemical reactions needed to stay alive, and to pass
information for maintaining life onto the next generation.
\section*{Background}
\subsection*{DNA, RNA and protein in general}
The name of the process when DNA goes from DNA to RNA to protein is called the \textbf{Central Dogma}. I will focus on
the cell, how we perceive it and the interaction inside it. The cell can be seen as a network community of
interacting protein molecules. The protein comes from our DNA.

In our DNA, there are areas that contain codes for making protein. These areas are called genes. Also, all of the
possible interactions between the cells are specified by the proteins in complex social networks. The reason for calling
these networks for "social networks" is because it is the interactions that we look at as the connections between the
cells. It is our protein that performs chemical reactions in our body. DNA on the other hand stores and passes
information about how our body is built up. RNA is the intermediate stage between DNA and protein.

\subsection*{Protein}
The protein in our body is made up from amino acids. The amount of amino acids in a protein may vary from 20 to 5000. But
on average there is about 350 amino acids in a protein in our body. The way we identify amino acids is through the
alphabet. We use almost every letter in the alphabet, and it is organized in a chronological order when we show
them in a list, but we miss some characters. They can also be identified by three letters, or the whole name of the
amino acid.

Amino acids have some very basic attributes like volume and mass. But beeing acids we also have information about their
polarity and their basicity/acidity. So the amino acids are either polar or non-polar, combined with beeing neutral,
acidic or basic. And of course we are able to see to what degree they are acidic/basic.

Amino acids consist of an amino group, carboxyl group and a R group. Very often there is also a central carbon that
combines all of these groups together. We have about 20 different amino acids and they can be classified into 4 types.
The positively charged, the negatively, the polar and the non-polar. The positive amino acids are basic and the negative
ones are acidic. The four types just mentioned are not totally exclusively to eachother, though an amino acid cannot be
both basic and acidic at the same time. At the same time an amino acid cannot be both polar and non-polar. But any
combination of acidity/basicity together with polarity can occur.
\subsection*{Genome}
When the genome changes suddenly and unexpected, we have a mutation. A mutation may happen when several things occur:
\begin{itemize}
    \item Insertion
        \begin{itemize}
            \item A part of a chromosome gets inserted into another one
        \end{itemize}
    \item Deletion
        \begin{itemize}
            \item A part of a chromosome gets deleted
        \end{itemize}
    \item Duplication
        \begin{itemize}
            \item A part of a chromosome gets duplicated
        \end{itemize}
    \item Inversion
        \begin{itemize}
            \item A part of a chromosome gets inverted, but yet it stays in place
        \end{itemize}
    \item Translocation
        \begin{itemize}
            \item Two chromosomes exchanges parts and they become what is called a derivative
        \end{itemize}
\end{itemize}
\printbibliography
\end{document}
