\documentclass[UKenglish,11pt,a4paper]{article}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc, url}
\usepackage{graphicx,babel,csquotes}
\usepackage[backend=biber,style=numeric-comp,sorting=nyt,maxbibnames=20,maxcitenames=20]{biblatex}
\addbibresource{macros.bib}
\addbibresource{references.bib}
\title{Ranklust - A bioinformatics solution to identify network biomarkers in cancer}
\author{Henning Lund-Hanssen}
\begin{document}
\maketitle
\tableofcontents{}
\part{Introduction}
To this day we still struggle with cancer. Even with all our modern equipment and knowledge we have still not been able
to tame this horrible disease. My thesis is about making a tool which gives cancer researchers an easier way of
identifying network biomarkers in cancer. 
\section{State of today}
\subsection{Biomarkers}
Biomarkers are at the centre of this thesis. They are what Ranklust should be able to detect and rank in order to
identify network biomarkers, and not just single molecules of them. A biomarker is a "biological measure of a biological
state" \cite{biomarker1}. It can be represented by the levels of a specific protein in our blood, a specific gene, or a
combination the two.

Biomarkers can be used for different purposes. They can be used to measure the effect of cancer drug treatment. That the
drug does what it is supposed to do. It can be used to predict disease development or the current stage of the
disease. Here is a list of what biomarkers currently can be used for:
\\\\
\textbf{Usages for biomarkers:} \cite{beyondpsa}
\begin{itemize}
    \item Disease disposition
        \begin{itemize}
            \item What is a patient's risk of developing cancer in the future?
        \end{itemize}
    \item Screening
        \begin{itemize}
            \item Does earlier detection of patients with cancer decrease morality?
        \end{itemize}
    \item Diagnostic
        \begin{itemize}
            \item Who has cancer? What is the grade of the cancer?
        \end{itemize}
    \item Prognostic
        \begin{itemize}
            \item What clinical outcome is most likely if therapy is not administered?
        \end{itemize}
    \item Predictive
        \begin{itemize}
            \item Which therapy is most appropriate?
        \end{itemize}
    \item Monitoring
        \begin{itemize}
            \item Was therapy effective? Did the patient's disease recur=
        \end{itemize}
    \item Pharmacogenomic
        \begin{itemize}
            \item What is the risk for adverse reaction to the prescribed therapeutic dose?
        \end{itemize}
\end{itemize}
\textbf{Characteristics for a good biomarker:}
\begin{itemize}
    \item Safe and easy to measure
    \item Cost efficient to follow up
    \item Modifiable with treatment
    \item Consistent across gender and ethnic groups
\end{itemize}
\subsection{Prostate specific antigen}
An example of a single molecule biomarker is the prostate specific antigen (PSA). This is a protein produced by the 
prostate gland in male humans. The identification of cancer with PSA is simple, the higher the level of PSA, measured in
ng/mL (nanograms per milliliter), the higher is the chance of the patient having prostate cancer \cite{cancerfacts}.

Today PSA is used for both identifying and evaluating the current stage of prostate cancer. This biomarker can be found 
by analyzing blood examples from patients, thus fulfilling some of the demands for a good biomarker,
but not all of them. It is easy to measure and easy to acquire, but not reliable enough to be used as the only marker
% find a better word, "tool" sucks!
to identify and determine the stage or remission of prostate cancer. The low grade of reliability comes from the fact
that even though higher levels of PSA shows higher chance of having prostate cancer, prostate cancer is not the only
reason to have elevated levels of PSA \cite{cancerfacts}.
% find other reasons? need to cite the cancerfacts twice?
Namely inflammation and enlargement of the prostate. Though, a man with both these cases may or may not develop prostate
cancer.

% Beyond PSA part


% PSA conclusion
% Part 1 of conclusion

% Part 2 of conclusion
So the conclusion for the PSA biomarker is that it is not reliable enough and are causing faulty treatment of prostate
cancer that may not even exist. Because even if a patient has prostate cancer, it may not be developing any further, 
promoting the case of not taking any action against it at all. So there is need for a new biomarker, or at least a 
better way of identifying them.

\subsection{Next-generation sequencing}
Today, rapid analyzing of genes and proteins are made available through Next-Generation Sequencing (NGS)\cite{ngs1}.
% Find a better source for this!
This opens up the possibility of looking at a bigger picture when trying to diagnose cancer patients. Through acquiring
more data, faster than before, there now exists databases with much information that is easy to access. This makes room
for building huge networks of proteins and genes, allowing for more extensively and thorough assays to be done. For
example, what if something that is classified as a prostate cancer biomarker only is viable when proteins that has not
been classified as a biomarker, also is present? Together they could represent a more appropriate
\emph{network biomarker}.
The amount of data that can be analyzed also opens up for another more personalized approach to each cancer patient.
Finding patient-specific biomarkers could make a huge impact on the quality of treatment \cite{personalized}.
\section{Next generation of biomarkers}
The PSA biomarker is over 20 years old \cite{psa-age}. Through those years, it has been discovered other and better
biomarkers for prostate cancer than PSA. Among those, PCA3, which is detectable through urine samples from patients. It
also has the benefit of not being affected by the size of the prostate gland \cite{pca3-size}. But still the results
could be better. Therefore, it has been tried to combine these two biomarkers in order to see if it is beneficial to
see the results from each biomarker in light of each other \cite{beyondpsa}. The results from these tests is that they
complement each other to a level of significance that makes it compelling to analyze them both to diagnose prostate
cancer. It is important to point out that even if these biomarkers are not the best at indicating if a patient as cancer
or not. Combined, these biomarkers are good at indicating progression and recurrence of prostate cancer.

But all of this is based on single genes or proteins. What if we looked at whole networks as biomarkers?
\section{Network}
\section{Clustering}
\section{Cytoscape}
% Rename the Cytoscape section to Software/Architecture/Development
% General description of development
Cytoscape is the open-source software platform the Ranklust app will be developed on. Its main purpose is to visualize
molecular interaction networks and biological pathways. It has an easy way to integrate \textbf{\textit{Apps}}, which may
be combined by other apps again to build big and complex applications which may solve problems in a bigger picture. The
goal of Ranklust is to cluster the networks we get from single gene prioritization and rank them in order to identify
network biomarkers. Apps taking care of making the networks already exists, but there still has to be made a decision
about whether or not they could be modified in order to better support the clustering.

% Databases
Which databases to use has to be considered. The reason to use databases is because they have information on how protein 
and genes form a network based on how they interact with eachother. The initial database candidates in Ranklust are
iRefIndex \cite{iri}, GeneMania \cite{gm} and STRING \cite{str}. These databases all have in common that it exists
Cytoscape apps made to use these databases. STRING however, does not have any repository available through the Cytoscape
app store, so interacting with the database through a new app in Cytoscape without making new plugins may be difficult.
On the other hand, both iRefIndex and GeneMania have their repositories easily available to the public together with
decent documentation. However, the difference between them is what they contain information about. iRefIndex contains
data about protein-protein interaction (PPI), while GeneMania contains data about genes.
% Check this next one out! Link from source!
Since proteins come from genes, GeneMania can also give us some information about proteins. Differences between the
two databases will be discussed in greater depth at a later stage.

The open-source plugins in Cytoscape to communicate with the databases are iRefScape \cite{iridb} for iRefIndex and 
GeneMANIA \cite{gmdb} for GeneMania.

% Cytoscape
Cytoscape is based on the Java programming language, which is a little bit untraditional for a software platform used to
develop bioinformatics tools. % cite needed!!!
The reason to choose Java above Python, Perl or other popular programming languages is simply because I am more
versed in Java programming than any other language. The Cytoscape community also promotes the idea of developing apps
that follow the \emph{OSGi} standard \cite{cytoscape-osgi}.

% OSGi and design patterns
Developing OSGi software should promote modularization % cite needed!!!
of the code and increase the probability of the app being launched as an official Cytoscape app, in addition to provide
other developers with the possibility of reusing my modules in their own apps. There exists several design patterns that
could prove to be useful in the development of Ranklust. Another strategy to follow may not be a direct design patter,
but more of a collection of them, is the clean code principles. More thorough examination of these strategies will
follow. % Cite needed for design pattern definition and clean code principles

% Prototyp section
Prototyping of different parts of the app will be done in Python and the Galaxy environment \cite{galaxy}. Python is an 
easy language to prototype with, and Galaxy is an easy environment to test small scripts with. Galaxy also has great 
logging of previous experiments combined with settings, so recreating simulations and comparing results is easy to do and
reliable. However, in my experience, Java comes up to par with Python in development speed once all of the boilerplate
code is written and a good deployment tool is used. Therefore the Cytoscape platform is used for the final deployment of
Ranklust.

% Maven section

\printbibliography
\end{document}
