\documentclass[a4paper,UKenglish]{ifimaster}
\usepackage[utf8]{inputenc}
\usepackage[UKenglish]{duomasterforside}
\usepackage[T1]{fontenc, url}
\usepackage{graphicx,babel,csquotes,textcomp,varioref,float}
\usepackage[backend=biber,style=numeric-comp,sorting=nyt,maxbibnames=20,maxcitenames=20]{biblatex}
\usepackage[toc]{glossaries}
\usepackage{listings}
\usepackage{xcolor}
\usepackage{parskip}
\usepackage{fancyvrb}
\urlstyle{sf}
\graphicspath{{images/}}
\addbibresource{macros.bib}
\addbibresource{references.bib}

\lstset{
    breaklines=true,
    frame=tb,
    numbers=left,
    stepnumber=1,
    numberstyle=\color{gray},
    keywordstyle=\color{blue},
    stringstyle=\color{red},
    commentstyle=\color{olive},
    keepspaces=true
}

\makeglossaries
\newglossaryentry{golden}
{
    name=golden standard,
    description={The prior scoring standard evolved from the DisGeNET and Dragon
    Database of Genes associated with Prostate Cancer}
}
\newglossaryentry{maa}
{
    name=MAA,
    description={The Multiple Attribute Additive method described to score
    clusters}
}
\newglossaryentry{mam}
{
    name=MAM,
    description={The Multiple Attribute Multiplicative method described to score
    clusters}
}
\newglossaryentry{pr}
{
    name=PR,
    description={The PageRank method described to score clusters}
}
\newglossaryentry{prwp}
{
    name=PRWP,
    description={The PageRank With Priors method described to score clusters}
}
\newglossaryentry{hits}
{
    name=HITS,
    description={The Hyperlink-Induced Topic Search method described to score
    clusters}
}
\newglossaryentry{maven}
{
    name=Maven,
    description={The Java build tool used to compile and build clusterMaker2 and
    Ranklust}
}
\newglossaryentry{ide}
{
    name=Integrated Development Environment,
    description={A development environment specialized in developing software
    products}
}
\newglossaryentry{mcl}
{
    name=MCL,
    description={Markov Cluster algorithm used in Cytoscape to cluster the
    networks}
}
\newglossaryentry{rsquared}
{
    name=R-squared,
    description={The coefficient of determination when talking about the grade
    of fitness of a linear regression}
}
\newglossaryentry{jensen}
{
    name=DISEASE,
    description={DISEASE database used for retrieving z-values, p-values and
    manually curated disease-gene associations}
}
\newglossaryentry{pipeline}
{
    name=pipeline,
    description={The time used from start to end when analyzing the network in
    Cytoscape}
\newglossaryentry{dragon}
{
    name=DDPC,
    description={Dragon Database of Genes associated with Prostate Cancer}
}
}

\title{Ranklust}
\subtitle{An extension of the Cytoscape clusterMaker2 plugin and its application
to find network biomarkers in prostate cancer}
\author{Henning Lund-Hanssen}

\begin{document}
\duoforside[program={Programming and Networks},
    dept={Department of Informatics},long]

\frontmatter{}

\setlength{\parskip}{12pt}
\setlength{\parindent}{12pt}

\chapter*{Abstract}
Next-generation technologies provide bioinformaticians with data on a scale
worthy of adding a new label to the biomedicine industry as a whole, namely
Big Data. With enormous amounts of data comes a great challenge as how to work
with it. Algorithmic approaches benefit greatly from a bigger size in the amount
of data to work with and has already shown to take great advantage of the data
that comes from high-throuput sequencing technologies. One of the advantages is
the potential to identify genes that promote diseases by analyzing networks
biologically related networks. Another advantage is the newly gained
understanding of the topological organization of large-scale molecular networks.
Clustering groups of these networks into subnetworks depending on their function
in the network is known as module detection in networks. The goal of this
assignment was to combine these two newly adapted methods in bioinformatics to
develop a tool in Cytoscape. This tool will rank clusters based on the score of
the nodes (genes) they contain. To exercise an exact experiment with this tool,
I performed an attempt at identifying prostate candidate cancer cluster
biomarkers through the use of this tool. The tool itself in Cytoscape was
developed in Java, and its task was to rank clusters based on the score of the
nodes in a network. Comparing the results of these ranked clusters was then
analyzed and plotted out through the use of Python scripts.


\tableofcontents{}
\listoffigures{}
\listoftables{}
\lstlistoflistings

\mainmatter{}

\include{intro}
\include{methods}
\include{results}
\include{conclusion}
\include{appendix}

\backmatter{}
\printglossaries
\printbibliography
\end{document}
